### PR TITLE
Revamped unittests for `tbl_survfit()` alongside, `add_n`, `add_nevent`, `add_p`

### PR DESCRIPTION
**Pull request by zdz2101**
**What changes are proposed in this pull request?**

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed: `renv::install()`
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: tests/testthat/test-tbl_survfit.R [hunk 1]
```diff
@@ -1,21 +1,166 @@
...
+  }
+
+  expect_silent(
+    trial |>
+      dplyr::mutate(dplyr::across(everything(), remove_labels)) |>
+      tbl_survfit(
+        include = c(trt, grade),
+        y = "Surv(ttdeath, death)",
+        times = 12,
+        label = list(trt = "TREATMENT", grade = "GRADE")
```
Comment (zdz2101) [inline tests/testthat/test-tbl_survfit.R:66]: @ddsjoberg only thing I've "found" is that if there's pre-existing labels like there is in `trial` rn, it won't work, not sure if that was the intended consequence

**Review by zdz2101 — COMMENTED** [review]
@ddsjoberg added test for I think each argument, and "common" errors, not sure if you wanted an/multiple thrown in there too?
```
expect_equal(
{tbl_survfit values},
{ard_survival_survfit values}
)
```

File: tests/testthat/test-tbl_survfit.R (hunk 1 above)
Comment (zdz2101) [reply tests/testthat/test-tbl_survfit.R:66]: It'll overwrite if its just a single value though

Comment (ddsjoberg) [reply tests/testthat/test-tbl_survfit.R:66]: I think this should be fixed now, thank you!!!

**Review by ddsjoberg — APPROVED** [review]
thanks @zdz2101 !!!!

**Pull request merged**